tooling/create-tauri-app/test/index.spec.ts

```diff
@@ -0,0 +1,165 @@
+    timeout: timeoutLong
+  })
+}, timeoutLittleLonger)
+
+describe('CTA', () => {
+  describe.each(recipes.map((recipe) => [recipe, 'tauri-app']))(
+    `%s recipe`,
+    (recipe: string, appName: string) => {
+      it(
+        'runs',
+        async () => {
+          // creates a temp folder to run CTA within (this is our cwd)
+          const folder = f.temp()
+          const appFolder = path.join(folder, appName)
+
+          // runs CTA with all args set to avoid any prompts
+          const cta = await execa(
+            'node',
+            [
+              ctaBinary,
+              '--manager',
+              manager,
+              '--recipe',
+              recipe,
+              '--ci',
+              '--dev'
+            ],
+            {
+              all: true,
+              stdio: logOut,
+              cwd: folder,
+              timeout: timeoutLong
+            }
+          )
+
+          // check to make certain it didn't fail anywhere
+          expect(cta.failed).toBe(false)
+          expect(cta.timedOut).toBe(false)
+          expect(cta.isCanceled).toBe(false)
+          expect(cta.killed).toBe(false)
+          expect(cta.signal).toBe(undefined)
+
+          // run a tauri build to check if what we produced
+          //  can actually create an app
+          //  TODO long term we will want to hook this up to a real test harness
+          //  and then run that test suite instead
+          let opts: string[] = []
+          if (manager === 'npm') {
+            opts = ['run', 'tauri', '--', 'build']
+          } else if (manager === 'yarn') {
+            opts = ['tauri', 'build']
+          }
+          const tauriBuild = await execa(manager, opts, {
+            all: true,
+            stdio: logOut,
+            cwd: appFolder,
+            timeout: timeoutLong
+          })
+
+          expect(tauriBuild.failed).toBe(false)
+          expect(tauriBuild.timedOut).toBe(false)
+          expect(tauriBuild.isCanceled).toBe(false)
+          expect(tauriBuild.killed).toBe(false)
+          expect(tauriBuild.signal).toBe(undefined)
+
+          const packageFileOutput: {
+            [k: string]: string | object
+          } = JSON.parse(
+            await fs.promises.readFile(
+              path.join(appFolder, 'package.json'),
+              'utf-8'
+            )
+          )
+          expect(packageFileOutput['name']).toBe(appName)
+
+          const assertCustom: { [k: string]: Function } = {
+            vanillajs: () => {
+              expect(packageFileOutput['scripts']).toMatchObject({
+                tauri: 'tauri'
+              })
+            },
+            reactjs: () => {
+              expect(packageFileOutput['scripts']).toMatchObject({
+                start: 'react-scripts start',
+                build: 'react-scripts build',
+                test: 'react-scripts test',
+                eject: 'react-scripts eject',
+                tauri: 'tauri'
+              })
+            },
```

Comment:
why not just check for `tauri` since that's the only thing we are adding and it makes sure the test won't fail if `create-react-app` decides to change its scripts

Reply:
They haven't changed the scripts for years so I wasn't worried. Now that refactored the code a bit and made this easier to do, it probably makes sense to do now just in case.